pkg/metrics/cluster/node.go

```diff
@@ -176,13 +175,6 @@ func (n *node) run() {
 func (n *node) performClusterReshard(ctx context.Context, joining bool) error {
 	if n.ring == nil || n.lc == nil {
 		level.Info(n.log).Log("msg", "node disabled, not resharding")
-		return nil
-	}
-
-	if n.cfg.ReshardTimeout > 0 {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeout(ctx, n.cfg.ReshardTimeout)
-		defer cancel()
```

Comment:
This isn't quite right. Now cluster wide reshards aren't subject to timeouts but refreshes are. This will cause the same problem we've had before but just with the other type of reshard.

The alternative here is this method initiated background (ie asynchronous, in a separate goroutine) reshards, so the cluster-wide reshard would exit immediately.

Reply:
(Or the Refresh method itself could determine a timeout, but still this method would take quite a long time to execute. I'm not sure that necessarily matters, though).

Reply:
Per discussion moved timeout to refresh function, and the node triggers the refresh in a background to minimize mutex locking.